### PR TITLE
lint: add no-misused promises to linter

### PR DESCRIPTION
**Pull request by bmzig**
The `no-misused-promises` rule enforces unresolved promises to be resolved before used (e.g. in a boolean conditional or in an object). Adding this in the linter will ideally prevent bugs which occur due to handling promises instead of the resolved value.

### CODE REVIEW

File: src/dataworker/Dataworker.ts [hunk 1]
```diff
@@ -2221,7 +2221,7 @@ export class Dataworker {
     // FIXME: Temporary fix to disable root cache rebalancing and to keep the
     //        executor running for tonight (2023-08-28) until we can fix the
     //        root cache rebalancing bug.
-    if (!this.rootCache[key] || process.env.DATAWORKER_DISABLE_REBALANCE_ROOT_CACHE === "true") {
+    if (!(await this.rootCache[key]) || process.env.DATAWORKER_DISABLE_REBALANCE_ROOT_CACHE === "true") {
```
Comment (james-a-morris) [inline src/dataworker/Dataworker.ts:2224]: This is meant to be unawaited - we're testing for the existence of the promise

**Review by james-a-morris — CHANGES_REQUESTED** [review]
I think some of these would impact dataworker timings

File: src/dataworker/Dataworker.ts (hunk 1 above)
Comment (bmzig) [reply src/dataworker/Dataworker.ts]: Is this true for the other changes? This one is fixed here: https://github.com/across-protocol/relayer-v3/pull/1581/commits/f67cd4156a12f8943643d21025bfd481bd9eb022

Comment (bmzig) [reply src/dataworker/Dataworker.ts]: Also fixed here: https://github.com/across-protocol/relayer/pull/1581/commits/b6826bfaa25c5027460cf1dd3d65a2ecbb4146d1

**Review by nicholaspai — APPROVED** [review]

**Comment by nicholaspai** [comment]
> I think some of these would impact dataworker timings

Agreed. Sometimes we purposefully don't await promises. We cache the unresolved promise so the promises can execute in the background

File: .eslintrc.js [hunk 2]
```diff
@@ -36,8 +36,9 @@ module.exports = {
     "@typescript-eslint/no-unused-vars": ["error", { ignoreRestSiblings: true }],
     "chai-expect/missing-assertion": 2,
     "no-duplicate-imports": "error",
-    //    "require-await": "error",
+    // "require-await": "error",
```
Comment (james-a-morris) [inline .eslintrc.js:39]: If this is wrong we can remove it

**Review by james-a-morris — APPROVED** [review]

File: .eslintrc.js (hunk 2 above)
Comment (bmzig) [reply .eslintrc.js]: https://github.com/across-protocol/relayer/pull/1581/commits/40a8061d238182a95e35a0cfb09af8486ee28ac5

**Pull request merged**